src/main/java/games/strategy/engine/framework/GameRunner.java

```diff
@@ -656,4 +702,41 @@ public static void exitGameIfFinished() {
     });
   }
 
+  /**
+   * todo, replace with something better
+   * Get the chat for the game, or null if there is no chat.
+   */
+  public static Chat getChat() {
+    final ISetupPanel model = setupPanelModel.getPanel();
+    if (model instanceof ServerSetupPanel) {
+      return model.getChatPanel().getChat();
+    } else if (model instanceof ClientSetupPanel) {
+      return model.getChatPanel().getChat();
+    } else {
+      return null;
+    }
+  }
+
+  public static boolean hasChat() {
+    return getChat() != null;
+  }
+
+  /**
+   * After the game has been left, call this.
+   */
+  public static void clientLeftGame() {
+    if (!SwingUtilities.isEventDispatchThread()) {
+      SwingAction.invokeAndWait(GameRunner::clientLeftGame);
+      return;
+    }
+    // having an oddball issue with the zip stream being closed while parsing to load default game. might be caused by
+    // closing of stream while unloading map resources.
+    ThreadUtil.sleep(100);
+    setupPanelModel.showSelectType();
+    showMainFrame();
+  }
+
+  public static void quitGame() {
+    mainFrame.dispatchEvent(new WindowEvent(mainFrame, WindowEvent.WINDOW_CLOSING));
```

Comment:
Yeah, it's not as clear-cut to me anymore due to all the new EDT-bound methods that were added in the most recent commits (e.g. `createFileDialog()`, `showFileChooser()`, etc.).

I guess what was niggling me before was, as a dev, how do I know that `quitGame()` is required to run on the EDT?  The method name doesn't suggest anything related to the UI, and `GameRunner` is not in a UI-specific package, so I probably would have assumed I could just call it from anywhere.  As you said, I will find out during testing that I called it from the wrong thread and adjust appropriately. :smile:

However, I would argue that it is clear that the new methods you just added **are** UI specific since most of them have UI types in their signatures or their names are suggestive of UI behavior.  Therefore, I would think twice before calling them from a non-UI thread and would probably avoid threading issues from the beginning.

**Bottom line is you've convinced me that the code change to `quitGame()` is not necessary.**  Maybe a Javadoc stating that the method needs to be run from the EDT would be helpful, but I'm not suggesting it be done because I realize the recent changes to `GameRunner` are still a work in progress.  Based on your [earlier comment](https://github.com/triplea-game/triplea/pull/1855#issuecomment-307584978), it looks like the UI code will most likely be moved out of `GameRunner` to a UI-specific class in the future anyway.  That's good enough for me. :smile: